applications/OptimizationApplication/python_scripts/utilities/opt_line_search.py

```diff
@@ -59,3 +61,71 @@ def GetInfo(self) -> dict:
                 'scaled_step': self.step}
 
         return info
+    
+class BB_Step(object):
```

Comment:
the class name is a bit too terse for my taste, and it should be CamelCase (without the underscore). Also, I think you don't have to derive from `object` in python 3.x anymore.

Reply:
removed